plugin.js

```diff
@@ -39,10 +39,10 @@ function plugin (fastify, options, pluginRegistrationDone) {
     )
   }
 
-  fastify.decorateRequest('session', getSession())
+  fastify.decorateRequest('session', null)
```

Comment:
I'm not sure this is correct. I am fairly certain I made it a function invocation so that the actual session object is retrieved each time `req.session` is used. In particular, the session object returned by `getSession` is proxied so that we can know when the session has been updated.

Reply:
Issue is not the `getter` it's the `setter`.